exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/metric_exporter/__init__.py

```diff
@@ -202,7 +209,132 @@ def export(
+        split_resource_metrics: List[ResourceMetrics] = []
+
+        for resource_metrics in metrics_data.resource_metrics:
+            split_scope_metrics: List[ScopeMetrics] = []
+            split_resource_metrics.append(
+                ResourceMetrics(
+                    resource=resource_metrics.resource,
+                    schema_url=resource_metrics.schema_url,
+                    scope_metrics=split_scope_metrics,
+                )
+            )
+            for scope_metrics in resource_metrics.scope_metrics:
+                split_metrics: List[Metric] = []
+                split_scope_metrics.append(
+                    ScopeMetrics(
+                        scope=scope_metrics.scope,
+                        schema_url=scope_metrics.schema_url,
+                        metrics=split_metrics,
+                    )
+                )
+                for metric in scope_metrics.metrics:
+                    split_data_points: List[DataPointT] = []
+                    split_metrics.append(
+                        self._create_metric_copy(
+                            metric=metric,
+                            data_points=split_data_points,
+                        )
+                    )
+
+                    for data_point in metric.data.data_points:
+                        split_data_points.append(data_point)
+                        batch_size += 1
+
+                        if batch_size >= self._max_export_batch_size:
+                            yield MetricsData(
+                                resource_metrics=split_resource_metrics
+                            )
+                            # Reset all the variables
+                            batch_size = 0
+                            split_data_points = []
+                            split_metrics = [
+                                self._create_metric_copy(
+                                    metric=metric,
+                                    data_points=split_data_points,
+                                ),
+                            ]
+                            split_scope_metrics = [
+                                ScopeMetrics(
+                                    scope=scope_metrics.scope,
+                                    schema_url=scope_metrics.schema_url,
+                                    metrics=split_metrics,
+                                )
+                            ]
+                            split_resource_metrics = [
+                                ResourceMetrics(
+                                    resource=resource_metrics.resource,
+                                    schema_url=resource_metrics.schema_url,
+                                    scope_metrics=split_scope_metrics,
+                                )
+                            ]
+
+                    if not split_data_points:
+                        # If data_points is empty remove the whole metric
+                        split_metrics.pop()
```

Comment:
Please add a test case where the batch max size is equal to the total amount of data points.